UnityProject/Assets/Scripts/Managers/SignalsManager.cs

```diff
@@ -0,0 +1,145 @@
+using System.Collections;
+using System.Collections.Generic;
+using Communications;
+using UnityEngine;
+using Mirror;
+using ScriptableObjects.Communications;
+using Random = System.Random;
+
+namespace Managers
+{
+	public class SignalsManager : NetworkBehaviour
+	{
+		private static SignalsManager signalsManager;
+		public static SignalsManager Instance
+		{
+			get
+			{
+				if (!signalsManager)
+				{
+					signalsManager = FindObjectOfType<SignalsManager>();
+				}
+
+				return signalsManager;
+			}
+		}
+
```

Comment:
```suggestion
	public class SignalsManager : SingletonManager<SignalsManager>
	{

```
Don't network managers, and use SingletonManager